webclient/app/src/app/commons/appTemplateDialog/AppTemplateAuthDialog.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from "react";
+import { Button, TextField, Container, DialogTitle, Typography, IconButton, Box } from '@mui/material';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import { useTranslation } from 'react-i18next';
+import AppTemplateRest from "../../services/AppTemplateRest";
+import { CloudSync, Close } from "@mui/icons-material";
+import ErrorAlert from "../alert/ErrorAlert";
+import AppTemplateDialogStyles from "./AppTemplateDialogStyles";
+import NotificationDialog from "../alert/NotificationDialog";
+
+function AppTemplateAuthDialog(props) {
+    const { appTemplate, handleRefresh, setOpenSuccessDialog } = props;
+    const appTemplateDialogStyles = AppTemplateDialogStyles();
+    const [downloadRequestData, setDownloadRequestData] = useState({ "appTemplateId": null, "username": "", "password": "" });
+    const [alert, setAlert] = useState({"open":false, "title": "ERROR", "message": ""});
+    const { t } = useTranslation();
+
+    const appTemplateRest = new AppTemplateRest();
+
+    const [openAuthDialog, setOpenAuthDialog] = useState(false);
+  
+
+    const handleChange = (event) => {
+        const { name, value } = event.target;
+        let downloadRequestDataNew = { ...downloadRequestData };
+        downloadRequestDataNew[name] = value;
+        setDownloadRequestData(downloadRequestDataNew);
+    }
+
+    const handleLogin = () => {
+        console.log("in handleLogin");
+        if (appTemplate.credentialsRequired) {
+            handleAlertClose();
+            setOpenAuthDialog(true);
+        } else {
+            handleAppTemplateReload();
+        }
+    }
+
+    const handleAppTemplateReload = () => {
+        downloadRequestData.appTemplateId = appTemplate.id;
+        appTemplateRest.updateTemplates(downloadRequestData).then(() => {
+            handleRefresh();
+            setOpenAuthDialog(false);
+            setOpenSuccessDialog(true);
+        }).catch(err => {
+            setAlert({"open":true, "title": t("alert.error"), "message": t(err.response.data.messageKey)});
+            console.log(err.response.data);
+        });
+    }
+
+    const handleAlertClose = () => {
+        setAlert({"open":false, "title": "ERROR", "message": ""});
+    }
+
+    const onClose = () => {
+        setDownloadRequestData({ "appTemplateId": null, "username": "", "password": "" });
+        setOpenAuthDialog(false);
+    }
+
+    return (
+        <Container>
+            <Button onClick={handleLogin} startIcon={<CloudSync />} >{t("button.loadtemplate")}</Button>
+            <NotificationDialog open={alert.open && !appTemplate.credentialsRequired} onClose={handleAlertClose} severity="error" title={t(alert.title)} message={t(alert.message)} />
```

Comment:
Please format this in the future that one prop is in one line. I think this is confusing to understand.